### PR TITLE
Rename Provider to TracerProvider

### LINKED ISSUE #198: Rename `Provider` to `TracerProvider`
We now have a `MeterProvider` so `Provider` by itself is ambiguous.

### PR DESCRIPTION
**Pull request by awiede**
Fixes #198 

Rename `Provider` to `TracerProvider` for both `api::Provider` and `sdk::Provider`.

### CODE REVIEW

File: opentelemetry-jaeger/src/lib.rs [hunk 1]
```diff
@@ -312,11 +312,11 @@ impl PipelineBuilder {
     }
 
     /// Build a configured `sdk::Provider` with the recommended defaults.
```
Comment (jtescher) [inline opentelemetry-jaeger/src/lib.rs:314]: May way to find all for stray comments like this one

Comment (TommyCpp) [inline opentelemetry-jaeger/src/lib.rs:314]: Should probably change this as well 
```suggestion
    /// Build a configured `sdk::TraceProvider` with the recommended defaults.
```

File: src/sdk/trace/provider.rs [hunk 2]
```diff
@@ -61,7 +61,7 @@ impl Provider {
     }
 }
 
-impl api::Provider for Provider {
+impl api::TracerProvider for TracerProvider {
     /// This implementation of `api::Provider` produces `sdk::Tracer` instances.
```
Comment (TommyCpp) [inline src/sdk/trace/provider.rs:65]:
```suggestion
    /// This implementation of `api::TraceProvider` produces `sdk::Tracer` instances.
```

**Comment by awiede** [comment]
Good catches @jtescher / @TommyCpp! I _think_ I got all the comments now.

**Pull request merged**